devito/types/sparse.py

```diff
@@ -140,6 +225,99 @@ def _dist_datamap(self):
         """
         return self.grid.distributor.glb_to_rank(self._support) or {}
 
+    @cached_property
+    def dist_origin(self):
+        return self._dist_origin
+
+    @cached_property
+    def _pos_symbols(self):
+        return [Symbol(name='pos%s' % d, dtype=np.int32)
+                for d in self.grid.dimensions]
+
+    @cached_property
+    def _point_increments(self):
+        """Index increments in each Dimension for each point symbol."""
+        return tuple(product(range(-self.r+1, self.r+1), repeat=self.grid.dim))
+
+    @cached_property
+    def _point_support(self):
+        return np.array(self._point_increments)
+
+    @cached_property
+    def _position_map(self):
+        """
+        Symbols map for the physical position of the sparse points relative to the grid
+        origin.
+        """
+        return OrderedDict([((c - o)/d.spacing, p)
+                            for p, c, d, o in zip(self._pos_symbols,
+                                                  self._coordinate_symbols,
+                                                  self.grid.dimensions,
+                                                  self.grid.origin_symbols)])
+
+    @cached_property
+    def _dist_reorder_mask(self):
```

Comment:
misplaced, group with sibling properties/methods

Reply:
likewise, there's a `_dist_scatter_mask` approx 60 lines below, and in the middle between the two, public methods

Reply:
up

Reply:
THis one is a cached property and groupped with it, the dist_scatter/gather are methods don't think can really move this one in a better place